hugegraph-core/src/main/java/com/baidu/hugegraph/util/StringEncoding.java

```diff
@@ -156,14 +153,13 @@ public static String encodeBase64(byte[] bytes) {
 
     public static String decompress(byte[] value) {
         BytesBuffer buf = BytesBuffer.allocate(value.length * 2);
-        try (ByteArrayInputStream bis = new ByteArrayInputStream(value);
-             GZIPInputStream in = new GZIPInputStream(bis)) {
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(value)) {
             byte[] bytes = new byte[value.length];
             int len;
-            while ((len = in.read(bytes)) > 0) {
+            while ((len = bis.read(bytes)) > 0) {
                 buf.write(bytes, 0, len);
             }
-            return decode(buf.bytes());
+            return decode(LZ4Util.decompress(buf.bytes(), BLOCK_SIZE).array());
```

Comment:
to avoid copy bytes, add method decode(bytes, offset, length), and let BytesBuffer implements OutputStream that  BytesBuffer instance could be pass into LZ4BlockOutputStream.

Reply:
address it:
use BytesBuffer instead of ByteArrayOutputStream in compress and decompress method, and init  BytesBuffer estimate size